From f7d99817e4720d1ebbd5149f2103e9e4a3069663 Mon Sep 17 00:00:00 2001
From: Weijie Gao <weijie.gao@mediatek.com>
Date: Mon, 8 Dec 2025 15:24:07 +0800
Subject: [PATCH 16/23] board: mediatek: fix FIT image kernel configuration
 detection

The FIT image kernel configuration may not have external data. The legacy
version uses embedded data as well.

Change-Id: I67fec5259b3416d27400df729c65ceb414e14e15
Signed-off-by: Weijie Gao <weijie.gao@mediatek.com>
CR-Id: WCNCR00180092
---
 board/mediatek/common/image_helper.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/board/mediatek/common/image_helper.c b/board/mediatek/common/image_helper.c
index 82b30625a4d..64b0afd2911 100644
--- a/board/mediatek/common/image_helper.c
+++ b/board/mediatek/common/image_helper.c
@@ -457,14 +457,12 @@ bool fit_image_check_kernel_conf(const void *fit, const char *conf)
 	if (noffset < 0)
 		return false;
 
-	ret = itb_external_image_data_info(fit, noffset, FIT_KERNEL_PROP,
-					   NULL, NULL);
-	if (ret)
+	ret = fit_conf_get_prop_node_index(fit, noffset, FIT_KERNEL_PROP, 0);
+	if (ret < 0)
 		return false;
 
-	ret = itb_external_image_data_info(fit, noffset, FIT_FDT_PROP,
-					   NULL, NULL);
-	if (ret)
+	ret = fit_conf_get_prop_node_index(fit, noffset, FIT_FDT_PROP, 0);
+	if (ret < 0)
 		return false;
 
 	return true;
-- 
2.45.2

